woocommerce/woocommerce-android
# Issue/3725 refactor card reader manager step 2

Parent issue #3725 

This PR
1. Renames "startDiscovery" to "discoverReaders"
2. Extracts logic related to reader discover into DiscoverReadersAction
3. "startDiscovery/discoverReaders" returns Flow now. It was a void method before and the consumer had to collect `discoveryEvents` MutableStateFlow. This change brings us two benefits - Firstly, it connects the producer with the consumer and secondly, we can easily cancel the discovery when the consumer stops collecting the flow.
4. Add tests

Note: CardReaderSettingsFragment is just a temporary class and doesn't need to be reviewed. It'll never be used in production.

To test - hw reader is NOT required
1. Open app settings
2. Click on "Manager card reader"
3. (optional) Check "use simulated card reader"
4. Tap on "Connect Card Reader"
5. Notice "Started" snackbar and notice state changes to "connecting/connected"

Update release notes:
